src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -52,24 +53,28 @@ public CommandBase spinDown() {
     }
 
     /*
-     * Calculates RPM -> velocity / circumference is how many rotation needed in a
-     * second, so times 60 gives us how many RPM we need. (returns inches/second)
-     * Also applies a 20% loss.
+     * Calculates RPM with some gear ratio mathematics. (returns inches/second) Also
+     * applies a 10% loss.
      */
     public static double calculateRPM() {
-        return 60 * calculateVelocity() / CIRCUMFERENCE * 0.8;
+        if (SmartDashboard.getBoolean("Sees Target", false)) {
+            return SmartDashboard.getNumber("Last RPM", 0);
+        } else {
+            final double RPM = calculateVelocity() * BALL_SPEED_RATIO * 0.9;
+            SmartDashboard.putNumber("Last RPM", RPM);
+            return RPM;
+        }
     }
 
     /*
-     * Finds the needed velocity to reach a target (x, y) or (distanceToTarget,
+     * Finds the needed velocity to reach a target (x, y) or (horizontalDistance,
      * verticalDistance). The formula takes takes theta or launch angle, target and
      * gravity.
      */
-
     public static double calculateVelocity() {
-        if (distanceToTarget * Math.tan(THETA) >= verticalDistance) {
-            final double gravitySide = GRAVITY * distanceToTarget * distanceToTarget;
-            final double tanSide = distanceToTarget * Math.tan(THETA) - verticalDistance;
+        if (horizontalDistance * Math.tan(THETA) >= verticalDistance) {
```

Comment:
Where is `horizontalDistance` being set? It looks like it's being set once in the constructor, and not updating. Also `verticalDistance` and `distanceToTarget`, which is honestly the one that you probably need updated regularly.

Reply:
verticalDistance is a constant otherwise yes i will do those